examples/flutter_gallery/lib/demo/material/full_screen_dialog_demo.dart

```diff
@@ -104,16 +106,15 @@ class FullScreenDialogDemoState extends State<FullScreenDialogDemo> {
   bool _allDayValue = false;
   bool _saveNeeded = false;
 
-  void handleDismissButton(BuildContext context) {
+  Future<bool> _onWillPop() async {
     if (!_saveNeeded) {
-      Navigator.pop(context, null);
-      return;
+      return true;
     }
 
     final ThemeData theme = Theme.of(context);
     final TextStyle dialogTextStyle = theme.textTheme.subhead.copyWith(color: theme.textTheme.caption.color);
 
-    showDialog<DismissDialogAction>(
+    return await showDialog<bool>(
```

Comment:
e.g. i think that would happen if you hit the back button while the dialog was up